pkg/ottl/parser.go

```diff
@@ -56,12 +56,28 @@ func (s *Statement[K]) Execute(ctx context.Context, tCtx K) (any, bool, error) {
 	return result, condition, nil
 }
 
-func NewParser[K any](functions map[string]interface{}, pathParser PathExpressionParser[K], enumParser EnumParser, telemetrySettings component.TelemetrySettings) Parser[K] {
-	return Parser[K]{
+func NewParser[K any](
+	functions map[string]interface{},
+	pathParser PathExpressionParser[K],
+	settings component.TelemetrySettings,
+	options ...Option[K],
+) Parser[K] {
+	p := Parser[K]{
```

Comment:
Could we set a default enum parser that returns an error saying that enums aren't supported for the current context? Right now if an enum parser option isn't given, the Collector will panic if a user tries to use enums in the context for that parser. This would also establish a pattern for future options.